test/fuzz/cbor/cbor.go

```diff
@@ -17,8 +17,90 @@ limitations under the License.
 package cbor
 
 import (
-	// Adds this package and its dependencies to the dependency chain of k8s.io/kubernetes
-	// without affecting the size of kube binaries. Once fuzz targets are added here, this will
-	// become a real import.
-	_ "k8s.io/apimachinery/pkg/runtime/serializer/cbor"
+	"fmt"
+	goruntime "runtime"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer/cbor"
+)
+
+var (
+	scheme      = runtime.NewScheme()
+	serializers = []cbor.Serializer{
+		cbor.NewSerializer(scheme, scheme),
+		cbor.NewSerializer(scheme, scheme, cbor.Strict(true)),
+	}
 )
+
+// FuzzDecodeAllocations is a go-fuzz target that panics on inputs that cause an unreasonably large
+// number of bytes to be allocated at decode time.
+func FuzzDecodeAllocations(data []byte) (result int) {
+	const (
+		MaxInputBytes     = 128
+		MaxAllocatedBytes = 16 * 1024
+	)
+
+	if len(data) > MaxInputBytes {
+		// Longer inputs can require more allocations by unmarshaling to larger
+		// objects. Focus on identifying short inputs that allocate an unreasonable number
+		// of bytes to identify pathological cases.
+		return -1
+	}
+
+	decode := func(serializer cbor.Serializer, data []byte) int {
+		var u unstructured.Unstructured
+		o, gvk, err := serializer.Decode(data, &schema.GroupVersionKind{}, &u)
```

Comment:
Decoding to unstructured makes sense here. But it does make me wonder... are there code paths we miss entirely by not decoding to structured types? I'm not suggesting we need to address that here, just trying to understand what exactly our coverage is.

Reply:
Yes, it can't reach 100% coverage. Other than branches for specific decoder configurations or library features that we don't use, there are some specialized branches that are only run for certain destination types that are never generated when decoding into unstructured, like time.Time.

Reply:
I added the interesting (coverage-increasing) inputs to the seed corpus in a separate commit. Looking through the coverage report, I don't see any uncovered branches that can be covered by decoding into unstructured. There are some branches that will become coverable as we opt in to some of the decode behaviors added in the most recent library version.